kinesis/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/KinesisCollector.scala

```diff
@@ -14,57 +14,41 @@
  */
 package com.snowplowanalytics.snowplow.collectors.scalastream
 
+import cats.effect.{IO, Resource}
+import com.snowplowanalytics.snowplow.collector.core.model.Sinks
+import com.snowplowanalytics.snowplow.collector.core.{App, Config}
+import com.snowplowanalytics.snowplow.collectors.scalastream.sinks.{KinesisSink, KinesisSinkConfig}
+import org.slf4j.LoggerFactory
+
 import java.util.concurrent.ScheduledThreadPoolExecutor
-import cats.syntax.either._
-import com.snowplowanalytics.snowplow.collectors.scalastream.generated.BuildInfo
-import com.snowplowanalytics.snowplow.collectors.scalastream.model._
-import com.snowplowanalytics.snowplow.collectors.scalastream.sinks.KinesisSink
-import com.snowplowanalytics.snowplow.collectors.scalastream.telemetry.TelemetryAkkaService
 
-object KinesisCollector extends Collector {
-  def appName      = BuildInfo.shortName
-  def appVersion   = BuildInfo.version
-  def scalaVersion = BuildInfo.scalaVersion
+object KinesisCollector extends App[KinesisSinkConfig](BuildInfo) {
+
+  private lazy val log = LoggerFactory.getLogger(getClass)
 
-  def main(args: Array[String]): Unit = {
-    val (collectorConf, akkaConf) = parseConfig(args)
-    val telemetry                 = TelemetryAkkaService.initWithCollector(collectorConf, BuildInfo.moduleName, appVersion)
-    val sinks: Either[Throwable, CollectorSinks] = for {
-      kc <- collectorConf.streams.sink match {
-        case kc: Kinesis => kc.asRight
-        case _           => new IllegalArgumentException("Configured sink is not Kinesis").asLeft
-      }
-      es         = buildExecutorService(kc)
-      goodStream = collectorConf.streams.good
-      badStream  = collectorConf.streams.bad
-      bufferConf = collectorConf.streams.buffer
-      sqsGood    = kc.sqsGoodBuffer
-      sqsBad     = kc.sqsBadBuffer
-      good <- KinesisSink.createAndInitialize(
-        kc.maxBytes,
-        kc,
-        bufferConf,
-        goodStream,
-        sqsGood,
-        es
+  override def mkSinks(config: Config.Streams[KinesisSinkConfig]): Resource[IO, Sinks[IO]] = {
+    val threadPoolExecutor = buildExecutorService(config.sink)
+    for {
```

Comment:
How about

```
for {
  stpe <- Resource.make(Sync[F].delay(buildExecutorService(config.sink))(stpe => Sync[F].delay(stpe.shutdown()))
  good <- ....
```

Reply:
Isn't executor closed [here](https://github.com/snowplow/stream-collector/blob/85723b9606bb56a439281a3032f6cee740351c30/kinesis/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/sinks/KinesisSink.scala#L349) as a part of sink shutdown?

Reply:
Ah yes.  That's fine then feel to ignore my comment.

If we were designing this from scratch then I would say better to shutdown the service very close in the code to where it is created.